### PR TITLE
Incoming user limits warnings

### PR DESCRIPTION
**Pull request by PClmnt**
## Description
This PR:

1. adds a banner that shows a user that is on the free plan that they are going to be user limited in X number of days. 
2. Adds an inline alert to the users page saying the same thing and linking them out to the plans page

N.B startDate hasn't been added to the license object yet so I tested with an epoch timestamp in the future, (which is where the `in X days ` comes from). That should be added before this PR is merged. 

## Screenshots
Screenshot 2023-04-21 at 13 44 29

image

### CODE REVIEW

File: packages/builder/src/components/portal/licensing/licensingBanners.js [hunk 1]
```diff
@@ -141,6 +145,36 @@ const buildPaymentFailedBanner = () => {
   }
 }
 
+const buildUsersAboveLimitBanner = EXPIRY_KEY => {
+  const userLicensing = get(licensing)
+  return {
+    key: EXPIRY_KEY,
+    type: BANNER_TYPES.WARNING,
+    criteria: () => {
+      return (
```
Comment (PClmnt) [inline packages/builder/src/components/portal/licensing/licensingBanners.js:154]: Important logic.

File: packages/builder/src/pages/builder/portal/users/users/index.svelte [hunk 2]
```diff
@@ -81,6 +86,13 @@
       width: "1fr",
     },
   }
+
+  $: showUserLimitAlert =
```
Comment (PClmnt) [inline packages/builder/src/pages/builder/portal/users/users/index.svelte:90]: Same important logic in here as well.

File: packages/builder/src/components/portal/licensing/licensingBanners.js (hunk 1 above)
Comment (PClmnt) [reply packages/builder/src/components/portal/licensing/licensingBanners.js:154]: (i removed the negation but forgot to remove the comment. Will sort.

File: packages/builder/src/pages/builder/portal/users/users/index.svelte [hunk 3]
```diff
@@ -237,6 +251,25 @@
     <Body>Add users and control who gets access to your published apps</Body>
   </Layout>
   <Divider />
+  {#if showUserLimitAlert}
+    <InlineAlert
+      type="error"
+      onConfirm={() => {
+        window.open("https://budibase.com/pricing/", "_blank")
```
Comment (PClmnt) [inline packages/builder/src/pages/builder/portal/users/users/index.svelte:258]: need to change this to check for account portal access and link to the upgrade page if so

File: packages/builder/src/components/portal/licensing/licensingBanners.js [hunk 4]
```diff
@@ -141,6 +145,35 @@ const buildPaymentFailedBanner = () => {
...
+        get(admin).cloud &&
+        userLicensing.license.plan.model === Constants.PlanModel.PER_USER &&
+        userLicensing.isFreePlan &&
+        userLicensing.quotaUsage.usageQuota.users > 5
+      )
+    },
+    message: `Free plan changes - Users will be limited to ${
```
Comment (Rory-Powell) [inline packages/builder/src/components/portal/licensing/licensingBanners.js:161]: Can just sub in `utils.capitalise(user.license.plan)` to make this banner agnostic of the free plan and determined completely by the backend

File: packages/builder/src/pages/builder/portal/users/users/index.svelte [hunk 5]
```diff
@@ -81,6 +88,14 @@
       width: "1fr",
     },
   }
+
+  // Per user check
+  $: showUserLimitAlert =
+    $admin.cloud &&
+    $licensing.isFreePlan &&
+    $licensing.license.plan.model === Constants.PlanModel.PER_USER &&
+    $licensing.quotaUsage.usageQuota.users > 5
```
Comment (Rory-Powell) [inline packages/builder/src/pages/builder/portal/users/users/index.svelte:97]: Can we combine this logic with the logic for the banner in the store, so we can say
```js
$licensing.warnUserLimit
```

File: packages/builder/src/components/portal/licensing/licensingBanners.js [hunk 6]
```diff
@@ -141,6 +145,35 @@ const buildPaymentFailedBanner = () => {
...
+      userLicensing.license.quotas.usage.static.users.value
+    } users ${dayjs(
+      userLicensing.license.quotas.usage.static.users.value.startDate
+    ).fromNow()}`,
```
Comment (Rory-Powell) [inline packages/builder/src/components/portal/licensing/licensingBanners.js:165]: What would be shown when we go past the start date?
Probably we should hide the banner at that point, as we'll start to deactivate then

File: packages/builder/src/components/portal/licensing/licensingBanners.js (hunk 1 above)
Comment (PClmnt) [reply packages/builder/src/components/portal/licensing/licensingBanners.js]: Ah right, I misunderstood when startDate was going to be added. Updated and simplified this!

**Review by Rory-Powell — APPROVED** [review]
Brilliant work peter!

**Pull request merged**